server/monitor-types/group.js

```diff
@@ -11,35 +11,69 @@ class GroupMonitorType extends MonitorType {
-            for (const child of children) {
-                if (!child.active) {
-                    // Ignore inactive childs
-                    continue;
-                }
-                const lastBeat = await Monitor.getPreviousHeartbeat(child.id);
-
-                // Only change state if the monitor is in worse conditions then the ones before
-                // lastBeat.status could be null
-                if (!lastBeat) {
-                    heartbeat.status = PENDING;
-                } else if (heartbeat.status === UP && (lastBeat.status === PENDING || lastBeat.status === DOWN)) {
-                    heartbeat.status = lastBeat.status;
-                } else if (heartbeat.status === PENDING && lastBeat.status === DOWN) {
-                    heartbeat.status = lastBeat.status;
+        if (children.length === 0) {
+            // Set status pending if group is empty
+            heartbeat.status = PENDING;
+            heartbeat.msg = "Group empty";
+            return;
+        }
+
+        let worstStatus = UP;
+        const downChildren = [];
+        const pendingChildren = [];
+
+        for (const child of children) {
+            if (!child.active) {
+                // Ignore inactive children
+                continue;
+            }
+
+            const label = child.name || `#${child.id}`;
+            const lastBeat = await Monitor.getPreviousHeartbeat(child.id);
+
+            if (!lastBeat) {
+                if (worstStatus === UP) {
+                    worstStatus = PENDING;
                 }
+                pendingChildren.push(label);
+                continue;
             }
 
-            if (heartbeat.status !== UP) {
-                heartbeat.msg = "Child inaccessible";
+            if (lastBeat.status === DOWN) {
+                worstStatus = DOWN;
+                downChildren.push(label);
+            } else if (lastBeat.status === PENDING) {
+                if (worstStatus !== DOWN) {
+                    worstStatus = PENDING;
+                }
+                pendingChildren.push(label);
             }
-        } else {
-            // Set status pending if group is empty
+        }
+
+        if (worstStatus === UP) {
+            heartbeat.status = UP;
+            heartbeat.msg = "All children up and running";
+            return;
+        }
+
+        if (worstStatus === PENDING) {
             heartbeat.status = PENDING;
-            heartbeat.msg = "Group empty";
+            heartbeat.msg = pendingChildren.length > 0 ? `Pending child monitors: ${pendingChildren.join(", ")}` : "Child monitor pending";
+            return;
+        }
+
+        heartbeat.status = DOWN;
+
+        const downList = downChildren.join(", ");
+        const pendingList = pendingChildren.join(", ");
+        let message = downList ? `Child monitors down: ${downList}` : "Child monitor down";
+
+        if (pendingChildren.length > 0) {
+            message += `; pending: ${pendingList}`;
         }
```

Comment:
I think this can be simplified, right?

```suggestion
        let message = `Child monitors down: ${downChildren.join(", ")}`;

        if (pendingChildren.length > 0) {
            message += `; pending: ${pendingChildren.join(", ")}`;
        }
```